adapters/relevantdigital/relevantdigital.go

```diff
@@ -0,0 +1,278 @@
+		return nil, &errortypes.BadInput{
+			Message: "imp.ext not provided",
+		}
+	}
+	relevantExt := openrtb_ext.ExtRelevantDigital{PbsBufferMs: DEFAULT_BUFFER_MS}
+	if err := json.Unmarshal(bidderExt.Bidder, &relevantExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: "ext.bidder not provided",
+		}
+	}
+	return &relevantExt, nil
+}
+
+func (a *adapter) buildEndpointURL(params *openrtb_ext.ExtRelevantDigital) (string, error) {
+	params.Host = strings.ReplaceAll(params.Host, "http://", "")
+	params.Host = strings.ReplaceAll(params.Host, "https://", "")
+	params.Host = strings.ReplaceAll(params.Host, RELEVANT_DOMAIN, "")
+
+	endpointParams := macros.EndpointTemplateParams{Host: params.Host}
+	return macros.ResolveMacros(a.endpoint, endpointParams)
+}
+
+func (a *adapter) buildAdapterRequest(prebidBidRequest *openrtb2.BidRequest, params []*openrtb_ext.ExtRelevantDigital) (*adapters.RequestData, error) {
+	newBidRequest, err := createBidRequest(prebidBidRequest, params)
+
+	if err != nil {
+		return nil, err
+	}
+
+	reqJSON, err := json.Marshal(newBidRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := a.buildEndpointURL(params[0])
+	if err != nil {
+		return nil, err
+	}
+
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     url,
+		Body:    reqJSON,
+		Headers: getHeaders(prebidBidRequest),
+	}, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	impParams, errs := getImpressionsInfo(request.Imp)
+	if len(errs) > 0 {
+		return nil, errs
+	}
+
+	bidRequest, err := a.buildAdapterRequest(request, impParams)
+	if err != nil {
+		errs = []error{err}
+	}
+
+	if bidRequest != nil {
+		return []*adapters.RequestData{bidRequest}, errs
+	}
+	return nil, errs
+}
+
+func getImpressionsInfo(imps []openrtb2.Imp) (resImps []*openrtb_ext.ExtRelevantDigital, errors []error) {
+	for _, imp := range imps {
+		impExt, err := getImpressionExt(&imp)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		resImps = append(resImps, impExt)
+	}
+	return
+}
+
+func getHeaders(request *openrtb2.BidRequest) http.Header {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("X-Openrtb-Version", "2.5")
+
+	if request.Device != nil {
+		if len(request.Device.UA) > 0 {
+			headers.Add("User-Agent", request.Device.UA)
+		}
+		if len(request.Device.IPv6) > 0 {
+			headers.Add("X-Forwarded-For", request.Device.IPv6)
+		}
+		if len(request.Device.IP) > 0 {
+			headers.Add("X-Forwarded-For", request.Device.IP)
+		}
+	}
+	return headers
+}
+
+func getMediaTypeForBidFromExt(bid openrtb2.Bid) (openrtb_ext.BidType, error) {
+	if bid.Ext != nil {
+		var bidExt openrtb_ext.ExtBid
+		err := json.Unmarshal(bid.Ext, &bidExt)
+		if err == nil && bidExt.Prebid != nil {
+			return openrtb_ext.ParseBidType(string(bidExt.Prebid.Type))
```

Comment:
`ParseBidType` checks for all supported type. But Relevant digital doesn't support audio type. Should add additional checks to make sure value returned from `openrtb_ext.ParseBidType(string(bidExt.Prebid.Type))` is one of the supported media format type mentioned in `static/bidder-info/relevantdigital.yaml`

```
func ParseBidType(bidType string) (BidType, error) {
	switch bidType {
	case "banner":
		return BidTypeBanner, nil
	case "video":
		return BidTypeVideo, nil
	case "audio":
		return BidTypeAudio, nil
	case "native":
		return BidTypeNative, nil
	default:
		return "", fmt.Errorf("invalid BidType: %s", bidType)
	}
}
```